tests/platform_helper/providers/test_environment_variable.py

```diff
@@ -0,0 +1,47 @@
+import os
+
+import pytest
+
+from dbt_platform_helper.platform_exception import PlatformException
+from dbt_platform_helper.providers.environment_variable import (
+    EnvironmentVariableProvider,
+)
+
+
+class TestEnvironmentVariableProvider:
+    def setup_method(self):
+        self.provider = EnvironmentVariableProvider()
+        self.env_var = "TEST_ENV_VAR"
+
+    # Unset self.env_var after each test. Avoid KeyError by setting None as default when self.env_var is not set.
+    def teardown_method(self):
+        os.environ.pop(self.env_var, None)
```

Comment:
More pythonic would be the following since you're not using the value that's popped:

```suggestion
        del os.environ[self.env_var]
```